### DISCUSSION TITLE
Help export PDF with images in VSCode

### DISCUSSION THREAD

**Discussion by spivakj** [marp-team/marp#530] [Q&A] [answered]
I'm having trouble using the VScode extension to export a PDF with images. I wish for the images to be embedded in the PDF file so that I can easily send the PDF to others, but they aren't showing up. In VScode in the directory `Project/` and my images are in `Project/Slides`. I'm inserting the images like `[![h:300](/Project/Slides/Image.png)]`. It's showing up in the preview, but not when I export in any way. I'm sorry if this has been asked before, but I really want to get marp to work!

**Comment by yhatt** [comment, marked as answer]
Try relative path from the location of Markdown file.

```diff
-![h:300](/Project/Slides/Image.png)
+![h:300](./Project/Slides/Image.png)
```

Image paths starting with `/` are absolute paths. They will be resolved from different file locations depending on the context.

Details:

- While previewing in VS Code, the absolute path is resolved from the VS Code project root.
- While exporting, the absolute path is resolved from the file system root. [^1]
- While viewing in exported HTML, the absolute path is resolved from URL root.

If you want stable path resolution, **recommend always to use relative paths**.

[^1]: If enabled `Markdown › Marp: Strict Path Resolution During Export` experimental setting in Marp for VS Code, you also can force the absolute path resolution as from the project root while exporting.
